### ISSUE TITLE
[bug] Invalid duration gets valid when wrapping/cloning

### ISSUE THREAD

**Issue by kearnmachine** [moment/moment#4323]
https://jsfiddle.net/kearnmachine/fLccczaa/1/

**Comment by marwahaha** [comment]
this seems related to https://github.com/moment/moment/issues/3260

**Comment by kishoreaoe** [comment]
![image](https://user-images.githubusercontent.com/6896519/34401124-826f6570-eba9-11e7-969f-6bf44b4ae997.png)

I am getting Invalid Date when i am trying to use below code:

var m = moment(AUGUST-2017); // Problem is occuring only in IE.
Could you please help me out with this issue.

**Comment by extra-mayo** [comment]
@kishoreaoe The problem is that you passed in an invalid argument to moment(). Wrap the argument in quotation marks:  
```javascript
var m = moment("AUGUST-2017");
```

**Comment by kishoreaoe** [comment]
@extra-mayo , yeap. understood. thanks

**Comment by marwahaha** [comment]
It seems that cloning durations, or creating new durations from existing durations should preserve validity. This seems like a bug to me.

**Comment by mikeweltejr** [comment]
@marwahaha I took a look at this and the issue is not exactly with cloning, the issue is within the Duration constructor. When you pass in NaN it uses this to determine validity but sets all data objects to 0. Now when you clone the object is in fact valid because it has 0 for all it's data properties. So I guess the question is should it use 0 for validity check when NaN is passed in? Or should it set the data to be something that will make it invalid. I don't think this is a cloning issue though as much as it's an issue with passing in NaN. If you think anything needs to be done on this let me know happy to put in a PR for it. See the line here: [https://github.com/moment/moment/blob/develop/moment.js#L2734](url)

**Comment by mikeweltejr** [comment]
Cool I’ll take a stab at this and submit a PR.

**Comment by ichernev** [comment]
Something simple that sets NaN for properties and/or specifically handles invalid durations in the constructor (to produce an invalid duration) should do the trick.

**Comment by mavkaann** [comment]
I'll try to fix.

**Comment by TokenStyle** [comment]
I can solve the bug, because PR is open for about 5 months

<details>
  <summary>Why this is a bug</summary>
  
  The state of NaN is stored in milliseconds, because IsNumber is being checked (which NaN passes, because it is a number [Why-does-typeof-NaN-return-number (stackoverflow)](https://stackoverflow.com/questions/2801601/why-does-typeof-nan-return-number) ).
  https://github.com/moment/moment/blob/07d88ae4d189e0fc5c2ea0ff09f7cc8309080409/moment.js#L3024-L3030

  Further, when creating a new Duration, the correctness of the recorded data is checked, and where NaN (in milliseconds) is false, and is replaced by 0.
  https://github.com/moment/moment/blob/07d88ae4d189e0fc5c2ea0ff09f7cc8309080409/moment.js#L2733-L2743

  As mentioned in the documentation [(momentjs.com/docs/#/parsing/is-valid/)](https://momentjs.com/docs/#/parsing/is-valid/) when validating invalid moments, all of its clones (i.e. using moment.clone() ) have to produce invalid moments. But in this case, all clones are validated.
</details>

**Comment by TokenStyle** [comment]
The bug has already been resolved in this #4883 Pull Request, it remains to do Merge (#4883) with `develop` branch, and close this Issue (#4323).

**Comment by marwahaha** [comment]
Fixed by 2.25.3, tested in 2.26.0
```
var m = moment.duration(NaN)
console.log(m.isValid())
// false debugger eval code:1:9
var wrapped = moment.duration(m);
console.log(wrapped.isValid())
// false debugger eval code:1:9
var clone = m.clone();
console.log(clone.isValid())
// false debugger eval code 1:9
```

Thanks for reporting this issue! 
Please re-open if you still have concerns.
